test/mempool-test.js

```diff
@@ -177,6 +177,56 @@ describe('Mempool', function() {
     }));
   });
 
+  it('should get spend coins and reflect in coinview', async () => {
+    const wallet = new MemWallet();
+    const script = Script.fromAddress(wallet.getAddress());
+    const dummyCoin = dummyInput(script, random.randomBytes(32));
+
+    // spend first output
+    const mtx1 = new MTX();
+    mtx1.addOutput(wallet.getAddress(), 50000);
+    mtx1.addCoin(dummyCoin);
+    wallet.sign(mtx1);
+
+    // spend second tx
+    const tx1 = mtx1.toTX();
+    const coin1 = Coin.fromTX(tx1, 0, -1);
+    const mtx2 = new MTX();
+
+    mtx2.addOutput(wallet.getAddress(), 10000);
+    mtx2.addOutput(wallet.getAddress(), 30000); // 10k fee..
+    mtx2.addCoin(coin1);
+
+    wallet.sign(mtx2);
+
+    const tx2 = mtx2.toTX();
+
+    await mempool.addTX(tx1);
+
+    {
+      const view = await mempool.getCoinView(tx2);
+      assert(view.hasEntry(coin1));
+    }
+
+    await mempool.addTX(tx2);
+
+    // we should not have coins available in the mempool for these txs.
```

Comment:
maybe a small comment explaining that

`getCoinView`: returns unspent coins only

`getSpentView`: returns spent + unspent coins